### PR TITLE
fix(maat): proofs

### PR DESCRIPTION
**Pull request by cernicc**
### Description

Fixes maat proofs and adds a script for future use.

### CODE REVIEW

File: runtime/src/configs/mod.rs
Comment (th7nder) [inline runtime/src/configs/mod.rs:467]: What's going on here?

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (th7nder) [inline cli/polka-storage-provider/client/src/commands/proofs.rs:246]: This number is magical, can we reference the pipeline code it's drawing from?

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (th7nder) [inline cli/polka-storage-provider/client/src/commands/proofs.rs:521]: But it's not [0u8; 32] on the testnet, right?

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (th7nder) [inline cli/polka-storage-provider/client/src/commands/proofs.rs:83]: We should propose the defaults that were hardcoded here, If I were a new user, I wouldn't know where to put it and which number needs to be after which one.

**Review by th7nder — CHANGES_REQUESTED** [review]

File: runtime/src/configs/mod.rs
Comment (cernicc) [reply runtime/src/configs/mod.rs:467]: TrailingZeroInput adds zeros to subject and then decodes that. In our case we are using empty slice for subject. That is why we get [0; 32] randomness when called. Before the slice was hashed. Now it's only decoded.

While fixing the test I found [TestRandomness](https://paritytech.github.io/polkadot-sdk/master/frame_support_test/struct.TestRandomness.html). The latest commit uses that directly. So that we don't need to support our implementation.

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (cernicc) [reply cli/polka-storage-provider/client/src/commands/proofs.rs:521]: Answered [here](https://github.com/eigerco/polka-storage/pull/600#discussion_r1853982716). It is now :D

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (cernicc) [reply cli/polka-storage-provider/client/src/commands/proofs.rs:83]: I thought that we will eventually remove those commands? The defaults were also specific for a single use case

**Comment by jmg-duarte** [comment]
Looks good to me, but lets wait until #599 gets merged

**Review by th7nder — APPROVED** [review]

File: runtime/Cargo.toml
Comment (th7nder) [inline runtime/Cargo.toml]: `"dep:frame-support-test"`?

**Review by th7nder — CHANGES_REQUESTED** [review]

File: runtime/Cargo.toml
Comment (cernicc) [reply runtime/Cargo.toml]: Only if testnet flag is enabled. The frame_support_test dependency is used. It is used because of [TestRandomness](https://paritytech.github.io/polkadot-sdk/master/frame_support_test/struct.TestRandomness.html)

File: cli/polka-storage-provider/client/src/commands/proofs.rs
Comment (aidan46) [inline cli/polka-storage-provider/client/src/commands/proofs.rs:82]: Is sector number and sector id used interchangeably? I am asking cause we use `SectorId` from [storage_proofs_core](https://docs.rs/storage-proofs-core/latest/storage_proofs_core/sector/struct.SectorId.html) and `SectorNumber` as our internal type.

Comment (cernicc) [reply cli/polka-storage-provider/client/src/commands/proofs.rs:82]: Currently, they are used interchangeably. A quick search over the workspace shows 38 results where `sector_id` is used. We should probably unify that.

**Review by th7nder — APPROVED** [review]

**Review by jmg-duarte — APPROVED** [review]
LGTM, good job!

**Pull request merged**
